### PR TITLE
executor: fix a bug of duplicate key update

### PR DESCRIPTION
**Pull request by jackysp**
When insert a row with on-duplicate-key update statement into a table which does not have any unique keys, the batch update operation miss a not-nil check.
PTAL @coocood @zz-jason

### CODE REVIEW

File: executor/write.go [hunk 1]
```diff
@@ -1225,8 +1225,12 @@ func (e *InsertExec) batchUpdateDupRows(newRows [][]types.Datum, onDuplicate []*
 
 // fillBackKeys fills the updated key-value pair to the dupKeyValues for further check.
 func (e *InsertExec) fillBackKeys(fillBackKeysInRow []keyWithDupError, handle int64) {
-	e.dupOldRowValues[string(e.Table.RecordKey(handle))] = fillBackKeysInRow[0].newRowValue
+	filled := false
```
Comment (coocood) [inline executor/write.go]: How about we check the length of `fillBackKeysInRows` and return if zero?

Comment (jackysp) [reply executor/write.go]: PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**